doc/sphinx/azhelpgen/azhelpgen.py

```diff
@@ -35,10 +34,11 @@ def make_rst(self):
 
         for help_file in help_files:
             is_command = isinstance(help_file, _help.CommandHelpFile)
-            yield '.. cli{}:: {}'.format('command' if is_command else 'group', help_file.command)
+            yield '.. cli{}:: {}'.format('command' if is_command else 'group', help_file.command if help_file.command else 'az') #it is top level group az if command is empty
             yield ''
             yield '{}:summary: {}'.format(INDENT, help_file.short_summary)
             yield '{}:description: {}'.format(INDENT, help_file.long_summary)
+            yield '{}:docsource: {}'.format(INDENT, help_file.doc_source)
```

Comment:
it looks like we would need to specify docsource in every help file.  I'd much rather have a simple check that switches on the first param in the help file path to determine the help file path.  Example (untested):
switches = {
    'ad': '<path to ad _help.py',
    'login': '<path to login _help.py file>',
    etc.
}
docsource = switches[help_file.command.split()[0]]